FOCUS #249: EffectiveCost review (#250)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/metrics/effectivecost.md

# Effective Cost

> The Effective Cost represents a charge inclusive of the impacts of all reduced rates and discounts, augmented with the amortization of upfront charges (one-time or recurring). This cost is denominated in the [Billing Currency](#billingcurrency). The Effective Cost is commonly utilized to track and analyze spending trends. The amortization of upfront charges portion of this metric's value should be proportional to the [Quantity In Pricing Unit](#quantityinpricingunit) and the time granularity of the data.

Reviewer: Suggestions only. Use all or part of the updates as you see fit.

```suggestion
The Effective Cost represents a charge inclusive of the impacts of all reduced rates and discounts, augmented with the amortization of relevant pre-paid fees (one-time or recurring) paid to cover future eligible charges. The amortized  portion included should be proportional to the [Quantity In Pricing Unit](#quantityinpricingunit) and the time granularity of the data. This cost is denominated in the [Billing Currency](#billingcurrency). The Effective Cost is commonly utilized to track and analyze spending trends.
```

Author: We've used the term `amortization of upfront charges (one-time or recurring)` in remaining cost and unit price definitions. Should we rephrase all occurrences to `amortization of relevant pre-paid fees (one-time or recurring)`? 
Do we need an additional PR to address this in the BilledCost spec? :)

Reviewer 2: Technically, not all amortized purchases will be "upfront" or "pre-paid". It might be more inclusive to say `amortization of relevant purchases`.

> Practitioners are faced with two main challenges with respect to this metric:

Reviewer 3: I understand the challenges are actually resolved thanks to the Effective cost.

Maybe:

< `Practitioners are faced with two main challenges with respect to this metric:`
> `This metrics resolves two challenges that are faced by practionners:`

> 1. Practitioners need to amortize upfront fees over the duration of the commitment and distribute those fees to the appropriate reporting groups (e.g. tags, resources).

Author: ```suggestion
1. Practitioners need to amortize relevant purchases over the duration of the commitment and distribute them to the appropriate reporting groups (e.g. tags, resources).
```

> 1. Practitioners need to amortize relevant purchases over the duration of the commitment and distribute them to the appropriate reporting groups (e.g. tags, resources).

Reviewer 4: ```suggestion
1. Practitioners need to amortize relevant purchases, such as upfront fees, over the duration of the commitment and distribute them to the appropriate reporting groups (e.g. tags, resources).
```

2. Many commitment-based discount constructs include a recurring expense for the commitment for every billing period and must distribute this cost to the resources using the commitment. This forces reconciliation between the initial commitment line item per period and the actual usage line items.

> The EffectiveCost column MUST be present in the billing data. This column MUST be a valid numeric value of type Decimal and MUST NOT contain null values. The aggregated EffectiveCost for a billing period MAY NOT  match the charge received on the invoice for the same billing period.

Reviewer 2: Removing an extra space.

nit: Does the added sentence provide any value? If it's a note for practitioners, then it should probably be above. If it's a requirement for providers (which is the purpose of this paragraph), then it doesn't commit them to anything. It just acknowledges that the numbers may not match. Am I missing the key takeaway? 
```suggestion
The EffectiveCost column MUST be present in the billing data. This column MUST be a valid numeric value of type Decimal and MUST NOT contain null values. The aggregated EffectiveCost for a billing period MAY NOT match the charge received on the invoice for the same billing period.
```

## Description

> A charge inclusive of the impacts of all reduced rates and discounts, augmented with the amortization of relevant purchases (one-time or recurring) paid to cover future eligible charges.

Reviewer 4: Again not a charge. EffectiveCost as a noun is not a charge.

Author: Using 'cost' instead of 'charge':
_A **cost** inclusive of the impacts ..._

### Concerning Granularity and Distribution of Recurring Fee

Providers should distribute the commitment purchase amount instead of including a line item at the beginning of a period so practitioners do not need to manually distribute the fee themselves.

### Concerning Amortization Approaches

> The amortization of the purchase should be amortized using a methodology determined by the provider that reflects the needs of their customer base and is proportional with the Quantity In Pricing Unit and the time granularity of the line item. Should a practitioner desire to amortize relevant purchases using a different approach, the practitioner can do so using the [Billed Cost](#billedcost) for the line item representing the initial purchase.

Reviewer 4: Is this true? Not all purchases include upfront fees.

Reviewer: Is it eligible purchases?

Author: Changed to: _**Eligible purchases** should be amortized using a methodology determined by the provider that_

Reviewer 4: LGTM
